Sundial CI target fails on main due to empty test selection (and reveals batch predict signature regression)

Summary:
Running the Sundial model test target on main fails in CI. The make target exits with pytest code 5 because no tests are selected. After running the full model suite in the Sundial venv, there is also a regression in batch prediction override compatibility.

Steps to reproduce:
1. Run:
```bash
make test-sundial
```
2. Observe:
```text
collected 37 items / 37 deselected / 2 skipped / 0 selected
make: *** [Makefile:16: test-sundial] Error 5
```
3. Run full model tests in the Sundial venv:
```bash
.venvs/sundial/bin/python -m pytest tests/models/ -v
```
4. Observe failures in predict_batch override tests with TypeError:
```text
TypeError: _predict_batch() takes 3 positional arguments but 4 were given
```

Expected behavior:
- The Sundial target should execute a meaningful subset of model tests and pass/fail based on real assertions.
- Batch prediction should remain backward-compatible for model overrides that implement _predict_batch(data, episode_col) when quantile levels are not requested.

Actual behavior:
- Sundial target uses a keyword filter that matches no tests, causing pytest exit code 5.
- predict_batch always forwards quantile_levels, breaking older override signatures and causing TypeError.

Impact:
- Main branch CI fails on the Sundial job.
- A compatibility regression is hidden by the empty selector unless full tests are run.

Proposed fix:
1. Update Sundial make target filter to run the Sundial-safe subset (exclude ttm/timesfm/chronos2 tests rather than requiring sundial in test names).
2. In predict_batch, call _predict_batch(data, episode_col) when quantile levels are None, and only pass quantile levels when explicitly requested.

Acceptance criteria:
1. make test-sundial runs selected tests (not zero selected) and exits successfully when tests pass.
2. Existing predict_batch override tests pass without changing override signatures.
3. No regressions in quantile-level batch prediction behavior.

--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ test-ttm:
 	$(VENVS_DIR)/ttm/bin/python -m pytest tests/models/ -v -k ttm
 
 test-sundial:
-	$(VENVS_DIR)/sundial/bin/python -m pytest tests/models/ -v -k sundial
+	$(VENVS_DIR)/sundial/bin/python -m pytest tests/models/ -v -k "not ttm and not timesfm and not chronos2"
 
 test-timesfm:
 	$(VENVS_DIR)/timesfm/bin/python -m pytest tests/models/ -v -k timesfm

--- a/src/models/base/base_model.py
+++ b/src/models/base/base_model.py
@@ -497,7 +497,12 @@ class BaseTimeSeriesFoundationModel(ABC):
         if not input_ids:
             return {}
 
-        results = self._predict_batch(data, episode_col, quantile_levels)
+        if quantile_levels is None:
+            # Backward-compatible dispatch for model overrides that implement
+            # _predict_batch(self, data, episode_col) without quantile_levels.
+            results = self._predict_batch(data, episode_col)
+        else:
+            results = self._predict_batch(data, episode_col, quantile_levels)
 
         missing = input_ids - set(results.keys())
         if missing:

--- a/tests/models/test_timesfm_config.py
+++ b/tests/models/test_timesfm_config.py
@@ -1,0 +1,18 @@
+"""Tests for TimesFM configuration validation."""
+
+import pytest
+
+pytest.importorskip("torch")
+
+from src.models.timesfm.config import TimesFMConfig
+
+
+class TestTimesFMConfigValidation:
+    def test_interval_mins_default_is_valid(self):
+        cfg = TimesFMConfig()
+        assert cfg.interval_mins == 5
+
+    @pytest.mark.parametrize("interval_mins", [0, -5])
+    def test_interval_mins_must_be_positive(self, interval_mins):
+        with pytest.raises(ValueError, match="interval_mins must be positive"):
+            TimesFMConfig(interval_mins=interval_mins)

--- a/tests/models/test_timesfm_config.py
+++ b/tests/models/test_timesfm_config.py
@@ -1,18 +1,28 @@
 """Tests for TimesFM configuration validation."""
 
+import importlib.util
+
 import pytest
 
-pytest.importorskip("torch")
+pytestmark = pytest.mark.skipif(
+    importlib.util.find_spec("torch") is None,
+    reason="requires torch",
+)
 
-from src.models.timesfm.config import TimesFMConfig
+
+@pytest.fixture
+def timesfm_config_cls():
+    from src.models.timesfm.config import TimesFMConfig
+
+    return TimesFMConfig
 
 
 class TestTimesFMConfigValidation:
-    def test_interval_mins_default_is_valid(self):
-        cfg = TimesFMConfig()
+    def test_interval_mins_default_is_valid(self, timesfm_config_cls):
+        cfg = timesfm_config_cls()
         assert cfg.interval_mins == 5
 
     @pytest.mark.parametrize("interval_mins", [0, -5])
-    def test_interval_mins_must_be_positive(self, interval_mins):
+    def test_interval_mins_must_be_positive(self, timesfm_config_cls, interval_mins):
         with pytest.raises(ValueError, match="interval_mins must be positive"):
-            TimesFMConfig(interval_mins=interval_mins)
+            timesfm_config_cls(interval_mins=interval_mins)

